internal/storage/rocksdb/store.go

```diff
@@ -441,9 +494,19 @@ func (rdbIter *iter) HasNext() bool {
 }
 
 func (rdbIter *iter) Next() ([]byte, []byte) {
-	defer rdbIter.rdbIter.Next()
 	key := toByteArray(rdbIter.rdbIter.Key())
 	val := toByteArray(rdbIter.rdbIter.Value())
+	expireTs, _ := getExpireTs(val)
+	if expireTs > 0 && expireTs < hlc.UnixNow() {
+		rdbIter.rdbIter.Next()
+		if rdbIter.HasNext() {
```

Comment:
Check if this expire check can be included inside the `HasNext` method itself. So that this method can avoid returning `nil, nil`. In that method, we can return `false` or simply skip.

Reply:
Fixed.